src/parser.yy

```diff
@@ -25,11 +26,36 @@
 
 %code {
 # include "driver.h"
-void AddToSpeciesVector(std::vector<specie> &v1, const std::vector<specie> &v2) {
+
+template <class T>
+void MergeVectors(std::vector<T> &v1, const std::vector<T> &v2) {
 	for (const auto &i : v2) {
 		v1.push_back(i);
 	}
 }
+
+void MakeComposition(driver &drv, const std::string &moduleName, std::vector<specie> inputs, std::vector<specie> outputs) {
+	Module *module = &drv.modules.at(moduleName);
+	speciesMapping inputMapping;
+	speciesMapping outputMapping;
+
+	if (inputs.size() != module->inputSpecies.size()) {
+		throw CompositionException(moduleName, "input", inputs.size(), module->inputSpecies.size());
+	}
+	for (int i = 0; i < module->inputSpecies.size(); i++) {
+		inputMapping.insert(std::make_pair(module->inputSpecies[i], inputs[i]));
+	}
+
```

Comment:
what is the difference between module->inputSpecies and inputs?

Reply:
`module->inputSpecies` is the `inputSpecies` field of the Module `module`, while `input` is the arguments to the function `MakeComposition`.